manifests/server.pp

```diff
@@ -94,6 +94,6 @@
 
   Class['prometheus::install']
   -> Class['prometheus::config']
-  -> Class['prometheus::run_service']
+  -> Class['prometheus::run_service'] # Note: config must *not* be configured here to notify run_service.  Some resources in config.pp need to notify service_reload instead
```

Comment:
My first thought was to add
```puppet
if $restart_on_change {
  Class['prometheus::config'] ~> Class['prometheus::run_service']
}
```
to this class.  This would have been a mistake.  Hopefully this comment will stop someone else having the same idea in the future! :)